src/diff.c

```diff
@@ -526,7 +484,10 @@ int git_diff_patchid(git_oid *out, git_diff *diff, git_diff_patchid_options *opt
 	if ((error = git_hash_ctx_init(&args.ctx)) < 0)
 		goto out;
 
-	if ((error = git_diff_foreach(diff, file_cb, NULL, NULL, patchid_line_cb, &args)) < 0)
+	if ((error = (git_diff_print(diff,
+				     GIT_DIFF_FORMAT_PATCH_ID,
+				     git_diff_patchid_print_callback__to_buf,
+				     &args))) < 0)
```

Comment:
I like your approach, it's a lot simpler than what I was doing before.

Reply:
Thanks